Backends/common/fonts.lisp

```diff
@@ -198,7 +198,9 @@ of letters specified in a separate kerning-table."))
 
 (defun parse-text-style* (style)
   "Returns complete text-style without NIL components and with numeric size."
-  (cond ((text-style-p style)
+  (cond ((device-font-text-style-p style)
+         style)
+        ((text-style-p style)
          (multiple-value-bind (family face size)
              (clim:text-style-components style)
            (if (and family face size (numberp size))
```

Comment:
I suggest something like this:

```lisp
(defun parse-text-style* (style)
  "Returns complete text-style without NIL components and with numeric size."
  (flet ((new-text-style (family face font-size)
           (let ((default *default-text-style*))
             (make-text-style (or family (text-style-family default))
                              (or face (text-style-face default))
                              (climb:normalize-font-size font-size)))))
    (cond ((device-font-text-style-p style)
           style)
          ((text-style-p style)
           (multiple-value-bind (family face size)
               (text-style-components style)
             (if (and family face size (realp size))
                 style
                 (new-text-style family face size))))
          ((null style)
           (parse-text-style* *default-text-style*))
          ((and (listp style) (alexandria:length= 3 style))
           (apply #'new-text-style style))
          (t
           (error "Invalid text style specification ~S." style)))))
```

Reply:
I've pushed change with something in a similar spirit.

Reply:
Why not put the repeated code into a local function?

Reply:
the repeated code is a single function call hence it doesn't save much to me. but it doesn't clutter the code either, so I've force-pushed version with flet. if you are satisfied with the branch please merge it.